eslint.config.mjs

```diff
@@ -0,0 +1,339 @@
+        "packages/cosmic-swingset/t[0-9]/**/*",
+        "packages/cosmic-swingset/t[0-9].*/**/*",
+        "packages/cosmic-swingset/node_modules/**/*",
+        "packages/cosmic-swingset/build/**/*",
+        // a3p-integration specific ignores
+        "a3p-integration/agoric-sdk/**/*",
+    ],
+}, {
+    // Include both .js and .ts files for casting package
+    files: ["packages/casting/**/*.{js,ts,mjs,cjs}"],
+}, {
+    // Include both .js and .ts files for cache package
+    files: ["packages/cache/**/*.{js,ts,mjs,cjs}"],
+},  ...fixupConfigRules(compat.extends(
+    "@agoric",
+    "plugin:ava/recommended",
+    "plugin:require-extensions/recommended",
+)), {
+    plugins: {
+        "@typescript-eslint": typescriptEslint,
+        prettier,
+        "require-extensions": fixupPluginRules(requireExtensions),
+    },
+
+    linterOptions: {
+        reportUnusedDisableDirectives: true,
+    },
+
+    languageOptions: {
+        parser: tsParser,
+        ecmaVersion: 5,
+        sourceType: "module",
+
+        parserOptions: {
+            useProjectService: true,
+
+            projectService: {
+                allowDefaultProject: ["*.js"],
+                defaultProject: "tsconfig.json",
+            },
+
+            tsconfigRootDir: __dirname,
+            extraFileExtensions: [".cjs"],
+        },
+    },
+
+    settings: {
+        jsdoc: {
+            mode: "typescript",
+        },
+    },
+
+    rules: {
+        "@typescript-eslint/no-unused-expressions": "off",
+        "@typescript-eslint/no-empty-object-type": "warn",
+        "@typescript-eslint/no-unnecessary-type-constraint": "warn",
+        "@typescript-eslint/no-unsafe-function-type": "warn",
+        "@typescript-eslint/no-wrapper-object-types": "warn",
+        "@typescript-eslint/ban-ts-comment": ["error", {
+            "ts-expect-error": false,
+            "ts-nocheck": false,
+        }],
+
+        "@typescript-eslint/no-floating-promises": "error",
+
+        "no-void": ["error", {
+            allowAsStatement: true,
+        }],
+
+        "ava/no-skip-test": "off",
+        "ava/use-test": "off",
+        "@jessie.js/safe-await-separator": "off", // Temporarily disabled due to ESLint 9.x compatibility issue
+
+        "jsdoc/check-tag-names": ["error", {
+            definedTags: [
+                "alpha",
+                "beta",
+                "category",
+                "categoryDescription",
+                "defaultValue",
+                "document",
+                "group",
+                "groupDescription",
+                "internal",
+                "privateRemarks",
+                "remarks",
+            ],
+        }],
+
+        "prettier/prettier": "warn",
+        "no-use-before-define": "off",
+    },
+}, {
+    files: [
+        "packages/*/src/**/*.js",
+        "packages/*/tools/**/*.js",
+        "packages/*/*.js",
+        "packages/wallet/api/src/**/*.js",
+    ],
+
+    rules: {
```

Comment:
I this set of rules used to have a name. Please ensure it does. 

I also see some of these repeated. Please DRY